compiler/src/dotty/tools/dotc/transform/ElimRepeated.scala

```diff
@@ -54,17 +54,17 @@ class ElimRepeated extends MiniPhase with InfoTransformer { thisPhase =>
                       |""".stripMargin,
               sym.sourcePos)
           else
-            addVarArgsForwarder(sym, isJavaOverride, hasAnnotation)
+            addVarArgsForwarder(sym, isJavaVarargsOverride, hasAnnotation)
       else if hasAnnotation
         report.error("A method without repeated parameters cannot be annotated with @varargs", sym.sourcePos)
     end
 
     super.transform(ref) match
-      case ref1: SymDenotation if ref1.is(Method) =>
+      case ref1: SymDenotation if ref1.is(Method, butNot=JavaDefined) =>
         val sym = ref1.symbol
-        val isJavaOverride = (ref1 ne ref) && overridesJava(sym) // (ref1 ne ref) avoids cycles
-        transformVarArgs(sym, isJavaOverride)
-        if isJavaOverride then
+        val isJavaVarargsOverride = (ref1 ne ref) && overridesJava(sym) // (ref1 ne ref) avoids cycles
```

Comment:
I think that comment can be dropped since it's not only about avoiding cycles but also about only selecting the correct overrides